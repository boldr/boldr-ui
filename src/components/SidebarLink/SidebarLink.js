/* @flow */
import React from 'react';
import NavLink from 'react-router-dom/NavLink';
import FontIcon from 'react-md/lib/FontIcons';
import classnames from 'classnames';

import { StyleClasses } from '../../theme/styleClasses';

type Props = {
  href: string,
  className: ?string,
  activeClassName: ?string,
  icon: string,
  text: string,
};
const BASE_ELEMENT = StyleClasses.SIDEBAR_LINK;
const SECOND_ELEMENT = StyleClasses.SIDEBAR_LINK_ACT;

const SidebarLink = (props: Props) => {
  const classes = classnames(BASE_ELEMENT, props.className);
  const secondaryClasses = classnames(BASE_ELEMENT, props.className);
  return (
    <li className="nav-item">
      <NavLink to={ props.href } className={ classes } activeClassName={ secondaryClasses }>
        <FontIcon>{ props.icon }</FontIcon> { props.text }
      </NavLink>
    </li>
  );
};

export default SidebarLink;
