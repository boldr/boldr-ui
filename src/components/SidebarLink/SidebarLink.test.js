import React from 'react';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import SidebarLink from './SidebarLink';

describe('<SidebarLink />', () => {
  const wrapper = shallow(<SidebarLink text="widget" href="/asf" icon="cat" />);
  it('renders <SidebarLink /> without breaking', () => {
    expect(wrapper.find('li').length).toBe(1);
  });
  it('renders snapshot', () => {
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
