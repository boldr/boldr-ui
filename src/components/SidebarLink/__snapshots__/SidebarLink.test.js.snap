// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SidebarLink /> renders snapshot 1`] = `
<li
  className="nav-item"
>
  <NavLink
    activeClassName="boldrui-sidebar-link"
    className="boldrui-sidebar-link"
    to="/asf"
  >
    <FontIcon
      iconClassName="material-icons"
    >
      cat
    </FontIcon>
     
    widget
  </NavLink>
</li>
`;
